src/android/UnityAdsPlugin.java

```diff
@@ -1,99 +1,162 @@
 package com.platogo.cordova.unityads;
 
+import com.unity3d.ads.IUnityAdsLoadListener;
 import com.unity3d.ads.UnityAds;
-import com.unity3d.ads.mediation.IUnityAdsExtendedListener;
+import com.unity3d.ads.IUnityAdsInitializationListener;
+import com.unity3d.ads.IUnityAdsShowListener;
 import com.unity3d.ads.metadata.PlayerMetaData;
+import com.unity3d.ads.UnityAdsShowOptions;
 
 import org.apache.cordova.*;
 import org.json.JSONArray;
 import org.json.JSONException;
+
 import android.app.Activity;
 import android.content.Context;
 import android.util.Log;
 
 public class UnityAdsPlugin extends CordovaPlugin {
     private CallbackContext callbackID;
     private static final String TAG = "UnityAds";
-    private boolean isInitialized = false;
 
+    private AdsListener adsListener = new AdsListener();
+
+    public static String getVideoPlacementId(JSONArray args) {
+        try {
+            return args.getString(1);
+        } catch (JSONException e) {
+            Log.w(TAG, "get videoAdPlacementId failed" + e.getMessage());
+            return null;
+        }
+    }
 
     @Override
     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
         super.initialize(cordova, webView);
     }
-    
+
     @Override
     public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
```

Comment:
throws JSONException can be removed

```suggestion
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext)  {
```